engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala

```diff
@@ -968,52 +968,9 @@ class IrToTruffle(
                     ) =>
                   val tpe =
                     mod.unsafeAsModule().getScope.getTypes.get(tp.name)
-                  val any         = context.getBuiltins.any
```

Comment:
Removing all the specialized branch nodes and replacing them with a single construct is a magnificent improvement regardless of any other comments.